web/src/components/Incidents/AlertsChart/AlertsChart.tsx

```diff
@@ -50,19 +50,38 @@ const AlertsChart = ({ theme }: { theme: 'light' | 'dark' }) => {
     (state: MonitoringState) => state.plugins.mcp.incidentsData.incidentsActiveFilters,
   );
   const { i18n } = useTranslation();
+
+  const selectedGroupId = incidentsActiveFilters.groupId?.[0];
+
+  // Only show alerts data if it belongs to the currently selected incident
+  // This prevents showing previous incident's data during the transition
+  const displayAlertsData = useMemo(() => {
+    // If no incident is selected, show empty
+    if (!selectedGroupId) {
+      return [];
+    }
+
+    // If we're loading and have no data, show empty (prevents old data from showing)
+    if (alertsAreLoading && isEmpty(alertsData)) {
+      return [];
+    }
+
+    return alertsData;
+  }, [alertsData, alertsAreLoading, selectedGroupId]);
+
   // Use dynamic date range based on actual alerts data instead of fixed chartDays
   const dateValues = useMemo(() => {
-    if (!Array.isArray(alertsData) || alertsData.length === 0) {
+    if (!Array.isArray(displayAlertsData) || displayAlertsData.length === 0) {
```

Comment:
it seems displayAlertsData is always an array, would be better to use the correct type when defining the const